src/rustllvm/PassWrapper.cpp

```diff
@@ -1099,6 +1104,52 @@ LLVMRustPrepareThinLTOImport(const LLVMRustThinLTOData *Data, LLVMModuleRef M) {
   return true;
 }
 
+/// Converts the LLVMRustThinLTOData::ImportLists map into a nested list. The
+/// first level is a null-terminated array with an entry for each module. Each
+/// entry is a pointer that points to a null-termined array of module names. The
+/// first entry is always the name of the *importing* module, the following
+/// entries are  the names of the modules it imports from. Each module name is
+/// a regular C string.
```

Comment:
Maybe say "regular malloc'ed C string" or something along those lines, just so someone doesn't have to scan the code for those calls to `strndup` ?

Reply:
Will do!

Reply:
Also note: In the current version, the allocations here are just leaked. I should probably fix that.